### DISCUSSION TITLE
Packaging CefSharp winforms app as msix bundle

### DISCUSSION THREAD

**Discussion by smitelij** [cefsharp/CefSharp#4385] [General]
Hi,

I've been packaging CefSharp as a winforms app (.exe / application files) for the last year or so and it's been working great.
I had a request come in to package this instead as an msix bundle for better roll-out support. From Visual Studio this seems easy enough (add an msix packaging project to the solution). 

However, when trying to run this wrapper, it seems like CefSharp is missing all DLLs (AssertAllDependenciesPresent fails, and when turning that flag off I get general failure messages about not finding core cef dlls). Original winforms project still runs fine. Both projects build fine.

I've tried adding cefSharp dependencies to the msix packaging project (in addition to the core winforms project where they already are), but no luck there either. Thoughts or ideas on what might be going wrong?

Thank you in advance, and apologies, I am still a newbie Windows developer.

**Comment by smitelij** [comment]
Image A: screenshot of error when running the packaging project from Visual Studio
![image](https://user-images.githubusercontent.com/2637605/216638507-dc720cf4-31d8-49dd-8653-e8ef38bd3c22.png)

Image B: screenshot of error when running the installed application on machine (note- this is after turning off dependency checking)
![image](https://user-images.githubusercontent.com/2637605/216638681-917f34e5-7d1b-4397-89cc-213464a8326b.png)

**Comment by amaitland** [comment]
> it seems like CefSharp is missing all DLLs (AssertAllDependenciesPresent fails,

What is the exact error you are getting? (No images of exceptions please)

Which files are missing?

> and when turning that flag off I get general failure messages about not finding core cef dlls).

If the files are missing then an exception is expected. 

> I've tried adding cefSharp dependencies to the msix packaging project

Is your `msix` project using `packages.config` or `PackageReference` for `Nuget`?

If you are using `packages.config` then try adding the following to a PropertyGroup in your installer proj file before the .targets entries (which are at the bottom).

```xml
<CefSharpBuildAction>Content</CefSharpBuildAction>
```
